src/Update.cpp

```diff
@@ -1435,6 +1377,122 @@ void InfectiousToDeath(int cellIndex)
 	Cells[cellIndex].D++; //// one more dead person
 }
 
+// severity state functions
+
+void decrement(int& x) {x--;}
+void increment(int& x) {x++;}
```

Comment:
I would prefer that we use `std::function` and lambdas instead of function pointers.

Also I think this is an opportunity for some asserts/checks (that `x` never goes negative, either on the way down or on the way up due to overflow).

Reply:
Happy to add asserts for each severity state variable post-transition to check they are >= 0.
There are no asserts at present in CovidSim - do you want more of them going forward?

Reply:
I would :-). Let's use assert to begin with - we can think about moving to GSL's precond/postcond/invariant macros as we go forward.

Reply:
Done.